src/invidious/helpers/errors.cr

```diff
@@ -40,6 +31,24 @@ def error_template_helper(env : HTTP::Server::Context, status_code : Int32, exce
 
   issue_template += github_details("Backtrace", exception.inspect_with_backtrace)
 
+  return {issue_title, issue_template}
```

Comment:
Regarding the comment on lines  47/48, maybe try without the `{}`:
```suggestion
  return issue_title, issue_template
```

Reply:
Same problem occurs once you change

```crystal
  issue_template_components = get_issue_template(env, exception)
  issue_title, issue_template = issue_template_components
```
to
```crystal
  issue_title, issue_template = get_issue_template(env, exception)
```

Reply:
And what happens if you replace:
`def get_issue_template(...) : Tuple(String, String)`
with:
`def get_issue_template(...) : {String, String}`

Reply:
It still occurs.

The compiler error I mentioned isn't due to a compile time syntax or type error but rather a bug in the compiler itself.

```
BUG: `__temp_6137 = get_issue_template(env, exception)` at /home/syeopite/Work/Software/iv-org/invidious/src/invidious/helpers/errors.cr:49:33 has no type (Exception)
  from crystal in '??'
  from crystal in '??'
  from crystal in '??'
  (...)
  from crystal in '__crystal_main'
  from crystal in '??'
  from crystal in 'main'
  from /usr/lib/libc.so.6 in '??'
  from /usr/lib/libc.so.6 in '__libc_start_main'
  from crystal in '_start'
  from ???
Error: you've found a bug in the Crystal compiler. Please open an issue, including source code that will allow us to reproduce the bug: https://github.com/crystal-lang/crystal/issues
````

Reply:
https://github.com/crystal-lang/crystal/issues/15638